cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2253,7 +2279,7 @@ else()
 endif()
 
 if(ARROW_USE_XSIMD)
-  resolve_dependency(xsimd REQUIRED_VERSION "8.1.0")
+  resolve_dependency(xsimd REQUIRED_VERSIONS "9.0.0" "8.1.0")
```

Comment:
Do we have to maintain this list manually in the future? What do you think @cyb70289 ?

Reply:
Not sure of the problem but looks inconvenient to maintain the list.
Can we only support the latest version?
Is it necessary to also update xsimd version in [thirdparty/version.txt](https://github.com/apache/arrow/blob/master/cpp/thirdparty/versions.txt#L96)?

Reply:
Or can we support by default all versions above 8.1.0? If xsimd breaks compatibility, we can perhaps revisit.

Reply:
OK. I'll implement 8.1.0 or later condition.

FYI: I explain this case more:

* A CMake package can specify compatibility policy: https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#generating-a-package-version-file
* xsimd uses `SameMajorVersion` that means `8.1.0` isn't compatible with `9.0.0`
  * FYI: Our CMake package use `AnyNewerVersion` for convenient but `SameMajorVersion` may be better because we use semver: https://github.com/apache/arrow/blob/master/cpp/cmake_modules/BuildUtils.cmake#L443
* We can use `find_package()` to find a CMake package: https://cmake.org/cmake/help/latest/command/find_package.html
* We can specify required (compatible) versions by range to `find_package()` but it requires CMake 3.19: https://cmake.org/cmake/help/latest/command/find_package.html#basic-signature
  * > A version range with the format `versionMin...[<]versionMax` where versionMin and versionMax have the same format and constraints on components being integers as the single version. By default, both end points are included. By specifying `<`, the upper end point will be excluded. Version ranges are only supported with CMake 3.19 or later.
* But version range requires `versionMax`.
  * It means that we can't implement `AnyNewerVersion` behavior for a `SameMajorVersion` CMake package only with `find_package()` options.